### DISCUSSION TITLE
How do I retry to find elements or wait for something with "contains()" to exist?

### DISCUSSION THREAD

**Discussion by MatiDiez** [cypress-io/cypress#16949] [Questions and Help] [answered]
Hey everyone, I'm having a problem with my selectors. Basically my problem is that I have a couple of sections on my site that load after others. Specifically the header and the footer of the web app finish loading and then the page content is retrieved. The problem I'm having is that if I try to get an element from the page content that is also included in the header and the footer, It fails. 

To be more specific. I need to get a link that has the text "Create" so I'm doing. 

`cy.get('a').contains('Create')`

When the page finishes loading (meaning only the header and footer) the step is executed, it gets all links in both areas but the link I need is not loaded yet so it fails. 

What cypress is doing, is that it basically succeeded in getting all "a" elements, and within the timeout (4000) will look for the "contains" piece within those 'a' elements, so it will never find my button. 

My question is, how can I make cypress retry the whole selector as if it is one thing? Like literally, within 4 seconds, look through all links, and not the ones you already got?

**Comment by bahmutov** [comment, marked as answer]
Read https://on.cypress.io/retry-ability and watch https://www.youtube.com/watch?v=VSIaDbv4GyE

In your case just use `cy.contains('a', 'Create')` from https://on.cypress.io/contains

**Reply by MatiDiez** [reply]
I found that the reason that the get command was not retrying was because I had another command chained so I added an assersion in the middle and now it works. Thank you
